docs/developer/architecture/core/http-service.asciidoc

```diff
@@ -0,0 +1,42 @@
+[[http-service]]
+== HTTP service
+Allows plugins:
+
+* to extend the {kib} server with custom HTTP API.
```

Comment:
Probably we should merge docs by service type and document server- and client-side API inside?
When a user opens this doc, it's not obvious that client-side API is described in `client-side-services.asciidoc` file.

Reply:
There's some documentation we'll need to add. For example: we offer the uiSettings service client side but there isn't anything in the current developer docs about that.
For now, I'll just add some `NOTE` admonitions stating where the service is available for each service.